### PR TITLE
add class GridFSBucket

### LINKED ISSUE #4: GridFSBucket typings
Would love to see support for this class.  If I write one, should I file a pull request?

### PR DESCRIPTION
**Pull request by DrDanRyan**
This is my first time writing a .d.ts file and I was not sure about testing it, so please give it a look or test if you can.  Thanks!

Closes #4

### CODE REVIEW

File: mongodb.d.ts [hunk 1]
```diff
@@ -1352,3 +1352,53 @@ export interface CommandCursor extends Readable, NodeJS.EventEmitter {
     //http://mongodb.github.io/node-mongodb-native/2.1/api/CommandCursor.html#wrap
     wrap(stream: Stream): void;
 }
+
+// http://mongodb.github.io/node-mongodb-native/2.1/api/GridFSBucket.html
+export class GridFSBucket {
+  constructor(db: Db, options?: GridFSBucketOptions);
+  // http://mongodb.github.io/node-mongodb-native/2.1/api/GridFSBucket.html#delete
+  delete(id: string | ObjectID, callback: GridFSBucketErrorCallback): void;
```
Comment (CaselIT) [inline mongodb.d.ts]: The callback should be optional

File: mongodb.d.ts [hunk 2]
```diff
@@ -1352,3 +1352,53 @@ export interface CommandCursor extends Readable, NodeJS.EventEmitter {
...
+  // http://mongodb.github.io/node-mongodb-native/2.1/api/GridFSBucket.html#drop
+  drop(callback: GridFSBucketErrorCallback): void;
```
Comment (CaselIT) [inline mongodb.d.ts]: The callback should be optional

File: mongodb.d.ts [hunk 3]
```diff
@@ -1352,3 +1352,53 @@ export interface CommandCursor extends Readable, NodeJS.EventEmitter {
...
+  // http://mongodb.github.io/node-mongodb-native/2.1/api/GridFSBucket.html#find
+  find(filter?: any, options?: GridFSBucketFindOptions): Cursor;
+  // http://mongodb.github.io/node-mongodb-native/2.1/api/GridFSBucket.html#openDownloadStream
+  openDownloadStream(id: string | ObjectID, options?: Object): GridFSBucketReadStream;
+  // http://mongodb.github.io/node-mongodb-native/2.1/api/GridFSBucket.html#openDownloadStreamByName
+  openDownloadStreamByName(filename: string, options?: Object): GridFSBucketReadStream;
+  // http://mongodb.github.io/node-mongodb-native/2.1/api/GridFSBucket.html#openUploadStream
+  openUploadStream(filename: string, options: Object): GridFSBucketWriteStream;
```
Comment (CaselIT) [inline mongodb.d.ts]: options should be optional

File: mongodb.d.ts [hunk 4]
```diff
@@ -1352,3 +1352,53 @@ export interface CommandCursor extends Readable, NodeJS.EventEmitter {
...
+  // http://mongodb.github.io/node-mongodb-native/2.1/api/GridFSBucket.html#openUploadStreamWithId
+  openUploadStreamWithId(id: string | ObjectID, filename: string, options?: Object): GridFSBucketWriteStream;
+  // http://mongodb.github.io/node-mongodb-native/2.1/api/GridFSBucket.html#rename
+  rename(id: string | ObjectID, filename: string, callback: GridFSBucketErrorCallback): void;
```
Comment (CaselIT) [inline mongodb.d.ts]: The callback should be optional

**Comment by CaselIT** [comment]
Thank you for the PR. It looks good, I've added just a couple of comments to the files.

**Comment by DrDanRyan** [comment]
Ok, I made the suggested changes, also I realized that my interface defs should have semicolons at the end of each line instead of the commas that I had before.

**Comment by CaselIT** [comment]
Thanks, I'll merge it.

**Comment by CaselIT** [comment]
I'll add a pr to the registry with the update later today

**Pull request merged**